core/modules/rproc/InfileMerger.cc

```diff
@@ -185,19 +196,21 @@ bool InfileMerger::merge(std::shared_ptr<proto::WorkerResponse> response) {
 
     bool ret = false;
     // Add columns to rows in virtFile.
-    int resultJobId = response->result.jobid() * MAX_JOB_ATTEMPTS;
-    int attemptCount = response->result.attemptcount(); //
-    if (attemptCount >= MAX_JOB_ATTEMPTS) {
-        LOGS(_log, LOG_LVL_ERROR, queryIdJobStr << " Canceling query attemptCount too large at " << attemptCount);
-        return false;
+    int resultJobId = makeJobIdAttempt(response->result.jobid(), response->result.attemptcount());
```

Comment:
Could the JobID calculation moved into a method inside the result object:

  response->result.jobId();

...unless this is a PROTOBUF object.